addons/website_hr_recruitment/static/src/js/website_hr_applicant_form.js

```diff
@@ -25,8 +25,10 @@ publicWidget.registry.hrRecruitment = publicWidget.Widget.extend({
     _onClickApplyButton (ev) {
         const $linkedin_profile = $('#recruitment4');
         const $resume = $('#recruitment6');
-        if ($linkedin_profile.val().trim() === '' &&
-            !$resume[0].files.length) {
+
+        const is_linkedin_empty = !$linkedin_profile.length || ($linkedin_profile.length && $linkedin_profile.val().trim() === '')
+        const is_resume_empty = !$resume.length || ($resume.length && !$resume[0].files.length)
```

Comment:
since we will not go to the part right of || if the part at the left is true, this should be equivalent to:

```suggestion
        const is_linkedin_empty = !$linkedin_profile.length || $linkedin_profile.val().trim() === '';
        const is_resume_empty = !$resume.length || !$resume[0].files.length;
 ```
(also ; missed in these lines)

I think that we could even use `?.` operator (I think available in odoo since 16.3 at least) and just keep what was there before:

```suggestion
        const is_linkedin_empty = !$linkedin_profile.val()?.trim();
        const is_resume_empty =!$resume[0]?.files?.length;
```

Reply:
when using `!$linkedin_profile.val()?.trim()` instead. if the linkedin field doesn't exist, it would be evaluated to `undefined` which is equal to `False` and the `&&` clause will be evaluated to `False` without checking if the resume field has been filled or not

I think the first version you suggested is clearer IMHO. Let me know what you think tho

Reply:
all three ways looks good to me